src/ui/marker.js

```diff
@@ -409,6 +411,21 @@ export default class Marker extends Evented {
         return this;
     }
 
+    _occluded(unprojected: LngLat): boolean {
+        const map = this._map;
+        if (!map) return false;
+        const camera = map.getFreeCameraOptions();
+        if (camera.position) {
+            const cameraLngLat = camera.position.toLngLat();
+            const shortestDistance = cameraLngLat.distanceTo(unprojected);
+            const distanceToMarker = cameraLngLat.distanceTo(this._lngLat);
+            // In globe view only occlude if past ~100 km from the camera.
+            // This fixes an issue where points at the center would be considered occluded
+            return shortestDistance < distanceToMarker * 0.9 && (!map._usingGlobe() || distanceToMarker > 100000);
```

Comment:
What brought you to this decision and choice of 100km? It'd be helpful to understand the original issue with a video, maybe there's more to dig on that particular issue. Because I'm curious whether this solves the root cause of :

>This fixes an issue where points at the center would be considered occluded

Reply:
This is an edge-case that occurs occasionally when the marker is positioned exactly in the center of the screen. Without this check, `_occluded` returns true and the marker it will disappear while it remains in that position.

This happens because these calculations are performed in 2d. `cameraLngLat` is really the transform "center", or the LngLat at the center of the screen. We compare the distance between the marker and this LngLat to the `unprojected` LngLat at the same screenspace , i.e. the surface of the map closest to the screen at that point.

The issue is that when `cameraLngLat` and the marker's `_lngLat` are at almost the same position, both distances can be much smaller than usual, and occasionally the small error introduced by the coordinate conversion becomes significant. For instance:

```js

this.lngLat: { lng: 0, lat: 0 }
cameraLngLat: { lng: -0.003285431640620118, lat: 0.043815645468953335 }
unprojected: { lng: -0.0032854316412169737, lat: 0.021842993031256697 }
shortestDistance: 2443.2508516109006
distanceToMarker: 4885.761564076163
returns true

```
The simplest approach to solving this issue was to disable occlusion when these values are too small.
The 100km choice is arbitrary, but it serves the purpose of being large enough to stop the problem while small enough to not interfere with correct occlusion (Which in globe should occur when the point is behind the horizon).
The fix only applies to globe. This issue probably can also occur with terrain enabled, but again it's very rare as it requires precise positioning, and even then only sometimes occurs.

I realize my comment here is rather nondescriptive (and inaccurate) and will update it!

Reply:
I suspect that a similar accuracy issue is the reason why the comparison uses `distanceToMarker * . 09`, this issue is effectively as an edge-case where the margin of error is too high.

If I'm correct, a more robust fix would be to replace that multiplication with something like:
```js
return shortestDistance < distanceToMarker - forgivenessInMeters;
```
`forgivenessInMeters` would need to scale with zoom in order to work with terrain as well.

Reply:
Hmm, could we simply disable occlusion with markers then?

Terrain is not noticeable enough (due to the low footprint/elevation at these zooms) that this will have meaningful visual difference anyway, instead of adding this extra logic specific to globe.

Reply:
Sorry my comment was not clear enough. 

I was suggesting to disable terrain occlusion, not globe occlusion when markers are behind the globe. Of course, globe occlusion is necessary as you mentioned.

>I wrote this code under the assumption that terrain and active globe view are mutually exclusive. 

Terrain and globe can be used at the same time already, so they're not mutually exclusive. I thought we were doing terrain occlusion as well as globe occlusion at the same, so suggested to disable terrain occlusion since the footprint of terrain is not noticeable on globe usually. So my suggestion to disable it is what you already do, you can disregard.